internal/wrappers/features/step_defs_cpp/steps.hpp

```diff
@@ -54,15 +52,24 @@ struct KuzzleCtx {
   std::vector<string> string_array;
 
   notification_result *notif_result = NULL;
-  CustomNotificationListener *listener;
 };
 
-class CustomNotificationListener : public NotificationListener {
+class CustomNotificationListener {
+  private:
+    CustomNotificationListener() {
+      listener = [](const kuzzleio::notification_result* res) {
+        ScenarioScope<KuzzleCtx> ctx;
+        ctx->notif_result = const_cast<notification_result*>(res);
+      };
+    };
+    static CustomNotificationListener* _singleton;
   public:
-    virtual void onMessage(notification_result *res) {
-      ScenarioScope<KuzzleCtx> ctx;
-
-      ctx->notif_result = res;
+    std::function<void(const kuzzleio::notification_result*)> listener;
```

Comment:
Why aren't you using your custom type defined here: https://github.com/kuzzleio/sdk-go/pull/198/files#diff-9baad34025ebec67948479fd9b5ba89fR9

Reply:
Because NotificationListener is const, it's used for parameters. if I use this type then I have a problem when I want to assign the value to listener https://github.com/kuzzleio/sdk-go/pull/198/files#diff-d404ec3bbad822f6000afa3909b36366R60

Reply:
Couldn't we just remove the constness from the typedef, and use const modifier when declaring a instance that we want to be constant?